wire-runtime/src/main/java/com/squareup/wire/MessageAdapter.java

```diff
@@ -26,11 +26,14 @@
 import static com.squareup.wire.Preconditions.checkNotNull;
 
 public abstract class MessageAdapter<M> {
+  /** Returns the raw type of the messages this adapter can read and write. */
+  public abstract Class<?> messageType();
```

Comment:
Gotcha. I hadn't gotten that far down. I saw `Class<M>` returned in the override and thought it might've just been an oversight.